sockeye/train.py

```diff
@@ -298,6 +298,7 @@ def create_data_iters_and_vocabs(args: argparse.Namespace,
                                          else None for i in range(len(args.source_factors))]
             source_vocab_paths = [args.source_vocab] + source_factor_vocab_paths
             target_vocab_path = args.target_vocab
+            num_pointers = max_seq_len_source if args.attention_based_copying else 0
```

Comment:
Also: At inference time the maximum source length can be longer than the maximum source length at training time. Maybe I missed it, but I didn't see any code guarding against predicting outside the range of the vocabulary. How do you protect against this? (the logic for the maximum input length is in `models_max_input_output_length` in `inference.py`)

One way this could be achieved is to have the `InferenceModel.max_supported_seq_len_source` return a non-None value in case the model uses pointer networks.

Reply:
OK, I added a check in the model loading.